apps/sms/services/activity/service.js

```diff
@@ -0,0 +1,85 @@
+/* global threads */
+
+(function(exports) {
+  'use strict';
+
+  /**
+   * Name of the service that is responsible for managing activities.
+   * @type {string}
+   */
+  const SERVICE_NAME = 'activity-service';
+
+  /**
+   * Definition of all available private members.
+   * @type {Object}
+   */
+  const priv = Object.freeze({
+    service: Symbol('ActivityService.service'),
+    activityRequest: Symbol('ActivityService.activityRequest'),
+
+    onPostResult: Symbol('ActivityService.onPostResult'),
+    onPostError: Symbol('ActivityService.onPostError'),
+    onActivityRequest: Symbol('ActivityService.onActivityRequest')
+  });
+
+  var ActivityService = function() {
+    var service = this[priv.service] = threads.service(SERVICE_NAME);
+
+    service.method('postResult', this[priv.onPostResult].bind(this));
+    service.method('postError', this[priv.onPostError].bind(this));
+
+    exports.navigator.mozSetMessageHandler(
```

Comment:
I suspect system messages will likely be replaced by "something" in service worker, but don't know how soon we'll get it as well as support for system messages in iframes with different URLs, while we need to load conversation in separate iframe (with dedicated URL) now...